src/modules/roc_pipeline/receiver.cpp

```diff
@@ -146,11 +146,18 @@ bool Receiver::create_session_(const packet::PacketPtr& packet) {
         roc_log(LogError, "receiver: can't create session, unexpected non-udp packet");
         return false;
     }
+
+    if (packet->udp()->dst_addr == config_.repair_port.address) {
+        roc_log(LogInfo, "receiver: can't create session, unexpected repair packet");
+        return false;
+    }
+
     const packet::Address src_address = packet->udp()->src_addr;
 
-    core::SharedPtr<ReceiverSession> sess = new (allocator_) ReceiverSession(
-        config_.default_session, config_.sample_rate, src_address, format_map_,
-        packet_pool_, byte_buffer_pool_, sample_buffer_pool_, allocator_);
+    core::SharedPtr<ReceiverSession> sess = new (allocator_)
+        ReceiverSession(config_.default_session, config_.sample_rate, src_address,
+                        format_map_, packet->rtp()->payload_type, packet_pool_,
```

Comment:
We assume here that this is an RTP packet. Since currently all supported source packets are RTP it's ok, but we need to add a check to create_session_():

```
    if (!packet->rtp()) {
        roc_log(LogError, "receiver: can't create session, unexpected non-rtp packet");
        return false;
    }
```

Reply:
78b04f7aa7c4b841a68a7095f30c0f62123cc258